spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployerActuatorTests.java

```diff
@@ -0,0 +1,238 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.stream;
+
+
+import java.util.Collections;
+import java.util.concurrent.ForkJoinPool;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.cloud.dataflow.core.DefaultStreamDefinitionService;
+import org.springframework.cloud.dataflow.core.StreamRuntimePropertyKeys;
+import org.springframework.cloud.dataflow.registry.service.AppRegistryService;
+import org.springframework.cloud.dataflow.server.controller.NoSuchAppException;
+import org.springframework.cloud.dataflow.server.controller.NoSuchAppInstanceException;
+import org.springframework.cloud.dataflow.server.repository.StreamDefinitionRepository;
+import org.springframework.cloud.deployer.spi.app.AppInstanceStatus;
+import org.springframework.cloud.deployer.spi.app.AppStatus;
+import org.springframework.cloud.deployer.spi.app.DeploymentState;
+import org.springframework.cloud.skipper.client.SkipperClient;
+import org.springframework.cloud.skipper.domain.ActuatorPostRequest;
+import org.springframework.cloud.skipper.domain.Info;
+import org.springframework.cloud.skipper.domain.Release;
+import org.springframework.cloud.skipper.domain.Status;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for the actuator operations on {@link SkipperStreamDeployer}.
+ *
+ * @author Chris Bono
+ */
+class SkipperStreamDeployerActuatorTests {
+
+	private SkipperClient skipperClient;
+
+	private SkipperStreamDeployer skipperStreamDeployer;
+
+	private final String releaseName = "flipflop3";
+	
+	private final String appId = "flipflop3.log-v1";
+
+	private final String instanceId = "flipflop3.log-v1-0";
+
+	private final String endpoint = "info";
+
+	@BeforeEach
+	void prepareForTest() {
+		skipperClient = mock(SkipperClient.class);
+		skipperStreamDeployer = new SkipperStreamDeployer(skipperClient, mock(StreamDefinitionRepository.class),
+				mock(AppRegistryService.class), mock(ForkJoinPool.class), new DefaultStreamDefinitionService());
+	}
+	
+	@Nested
+	class GetFromActuator {
+		@Test
+		void happyPath() {
+			Release release = newRelease(releaseName, appId, instanceId);
+			when(skipperClient.list(null)).thenReturn(Collections.singletonList(release));
+			skipperStreamDeployer.getFromActuator(appId, instanceId, endpoint);
+			verify(skipperClient).getFromActuator(releaseName, appId, instanceId, endpoint);
+		}
+
+		@Test
+		void withNoRegisteredApps() {
```

Comment:
**[Implementation Note]**: These tests are both checking the logic that is contained in the "determineReleaseName" method and could later be extracted out to a helper and thoroughly tested so that all the things that use it (actuator GET, actuator POST, and other methods currently in the stream deployer) do not have to also test it.